keps/sig-network/20180612-ipv4-ipv6-dual-stack.md

```diff
@@ -887,19 +894,19 @@ routed according to the families assigned for services.
 
 1. For Headless with selector, the value of `spec.ipFamilies` will drive the
    endpoint selection. Endpoints will use the value of `ipFamiles[0]`.
-   EndpointSlices will use endpoints with all IP families of the Service.
-2. For Headless without selector the values of `spec.preferDualStack` and
-   `spec.ipFamilies` are defaulted and saved as normal, but are ignored by
-   kubernetes (since the standard endpoint controllers ignore such services).
-
-> In all cases, the values of `spec.preferDualStack`, `spec.ipFamilies`, and
-> `spec.clusterIPs` are validated and must be consistent.
+   EndpointSlices will use endpoints with all IP families of the Service. This
+   value is expected to match cluster configuration. i.e. creating  service with
+   a family not configured on server will generate an error.
+2. For Headless without selector the values of `spec.ipFamilyPolicy` and
+   `spec.ipFamilies` are semantically validated for correctness but are not 
+   validated against cluster configuration. This means a headless service without
+   selectors can have IP families that are not configured on server.
```

Comment:
Need to explain the new defaulting for headless selectorless

Reply:
they are not defaulted. as per our agreement. or did i misunderstood that?

Reply:
Tim had said "DO change the defaulting for headless+selectorless to `PreferDualStack` and set `ipFamilies` to both" and I thought that that was the set of rules we had agreed on...

ie, that the default for headless+selectorless is dual-stack, but the user can override it explicitly, and we do expect consumers such as CoreDNS to obey `ipFamilies` in this case. (And the default value of "both" matches CoreDNS's legacy behavior of just creating both A and AAAA if there are dual-stack Endpoints.)

Reply:
This seems to be covered around line 594 now ?

Reply:
yes